File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    answer = 0
    width_max = 0
    height_max = 0
    for size in sizes:
        width, height = size
        width_max = max(width_max, width, height)
        height_max = max(height_max, min(width, height))
    answer = width_max*height_max
    return answer

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int width = 0;
        int height = 0;
        for (int[] s : sizes) {
            int a = Math.max(s[0], s[1]);
            int b = Math.min(s[0], s[1]);
            width = Math.max(width, a);
            height = Math.max(height, b);
        }
        return width * height;
    }
}